utils/helpers.ts

```diff
@@ -151,3 +153,56 @@ export function checkSearchTerm(searchTerm: string, plant: Plant) {
   // Check if plant_name contains searchTerm
   return plant.plant_name.toLowerCase().includes(searchTerm.toLowerCase());
 }
+
+export function fillCalendarGridArrayRowWithColor(
+  startMonth: string,
+  endMonth: string,
+  color: string,
+  rowIndex: number,
+  gridArray: string[],
+) {
+  // if startMonth and endMonth is both null, row should be empty
+  if (startMonth === null && endMonth === null) {
+    return gridArray;
+  }
+
+  // if either startMonth or endMonth are null, the null one should be set to the other
+  // this makes the time frame only one month long
+  if (startMonth === null) {
+    startMonth = endMonth;
+  } else if (endMonth === null) {
+    endMonth = startMonth;
+  }
+
+  // remove 'LATE' or 'EARLY' from startMonth and endMonth to query monthToIndex
+  const processedStartMonth = processPlantMonth(startMonth);
+  const processedEndMonth = processPlantMonth(endMonth);
+
+  // if the start month is LATE_MONTH, start column should be the second column for that month
+  // if the end month is EARLY_MONTH, end column should be the first column for that month
+  // otherwise, start column should be the first column and end column should be the second column for that month
+  const startColumn = startMonth.startsWith('LATE')
+    ? monthToIndex.get(processedStartMonth)! * 2 + 1
```

Comment:
I think we should remove the `!` operator here, and instead type monthToIndex as `Map<MonthEnum, number>`, defining the `MonthEnum` type in `schema.d.ts`

This way we can ensure that monthToIndex always returns a valid value.

This might be a tedious change, so we can also put this off.